### DISCUSSION TITLE
Unable to use "ctrl+shift+c" shortcut with Kitty to copy selected content to clipboard

### DISCUSSION THREAD

**Discussion by floating-cat** [helix-editor/helix#9457] [General]
### Summary

Kitty supports `ctrl+shift+c` for copying to clipboard: https://sw.kovidgoyal.net/kitty/overview/#other-keyboard-shortcuts.
But I when I use Helix with Kitty, the `ctrl+shift+c` doesn't work though `space+y` works fine.

I tried the same shortcut in Kitty without using Helix or in Kitty with VIM, it works fine.

### Reproduction Steps

 

I tried this:
1. `hx some-file-path`
2. Use the mouse to select some text.
3. Press `ctrl+shift+c`.

I expected this to happen:
The selected text is copied to the system clipboard.

Instead, this happened:
Nothing is is copied to the system clipboard.

### Helix log

```
2024-01-28T23:00:01.656 helix_view::clipboard [DEBUG] Using wl-copy+wl-paste to interact with the system and selection (primary) clipboard
2024-01-28T23:00:01.659 helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is supported in this terminal (checked in 3.147979ms)
2024-01-28T23:00:01.659 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2024-01-28T23:00:01.907 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-28T23:00:02.741 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2024-01-28T23:00:03.006 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2024-01-28T23:00:03.180 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
```

### Platform

Linux

### Terminal Emulator

Kitty 0.31.0

### Installation Method

nixpkgs

### Helix Version

helix 23.10 (f6021dd0)

**Comment by floating-cat** [comment]
I just noticed my case is similar to https://github.com/helix-editor/helix/discussions/5143
So maybe my issue is not a bug but work as intended?
But it is a little annoying I need to use the `ctrl+shift+c` in Kitty or some other Terminal emulator to copy the text to system clipboard but in Helix, I need to use the `space+y` instead.

Also, `ctrl+shift+v` works in Helix with Kitty so I think it is consistent to support `ctrl+shift+c` too in this case too.

**Comment by pascalkuthe** [comment]
helix selction has nothign to do with kippty concept of a selection so cs-c does not work.

the behaviour is the same for me in vim (if I select something in visual mode I can't copy it with cs-c no idea what you did there. cs-c is handled at the terminal emulator level its not a key event that is forwarded to us so we can't do anything with it. use <space>y or bind it to another key

**Reply by floating-cat** [reply]
If you open the VIM, use the mouse to select the some text. Using the ctrl+shift+v will work there. Though using the `v` and selecting indeed doesn't work in VIM either.
In helix, it seems it uses the select mode for both `v` and mouse selection, so it won't work for mouse selection.

Thanks for the info. I will bind it to another shortcut.

**Reply by pascalkuthe** [reply]
that is becsause we support mouse mode so mouse selects the helix selection but not vim by defualt. Yf you turn on mouse mode in vim it works the same there
